### PR TITLE
[Text Analytics] Use `TextAnalyticsAudience` in tests

### LINKED ISSUE #35603: [Text Analytics] Fix tests running in national clouds
Things to fix:
* Our test client always uses public cloud.
* Our pre-trained, static resource is currently only in public cloud.
* etc.

### PR DESCRIPTION
**Pull request by joseharriaga**
Resolves: https://github.com/Azure/azure-sdk-for-net/issues/35603

* Use `TextAnalyticsAudience` in tests so we can run them appropriately in the national clouds as needed.
* Ignore the tests that require our static resource in the national clouds.
* Add a check for AAD to `IsEnvironmentReadyAsync`
* Refactor existing tests.

### CODE REVIEW

File: sdk/textanalytics/tests.yml [hunk 1]
```diff
@@ -14,9 +14,6 @@ extends:
       Public:
         SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
         Location: '${{ parameters.Location }}'
-      Canary:
-        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
-        Location: 'centraluseuap'
```
Comment (heaths) [inline sdk/textanalytics/tests.yml:19]: Was this intentional? We provide needed backstop testing, so testing early against their early deployments is vital.

File: sdk/textanalytics/Azure.AI.TextAnalytics/perf/Azure.AI.TextAnalytics.Perf/Infrastructure/PerfTestEnvironment.cs [hunk 2]
```diff
@@ -26,13 +26,13 @@ private PerfTestEnvironment()
         /// <summary>
         /// The endpoint of the Text Analytics account to test against.
         /// </summary>
-        /// <value>The endpoint, read from the "TEXT_ANALYTICS_ENDPOINT" environment variable.</value>
-        public string Endpoint => GetVariable("TEXT_ANALYTICS_ENDPOINT");
+        /// <value>The endpoint, read from the "TEXTANALYTICS_ENDPOINT" environment variable.</value>
+        public string Endpoint => GetVariable("TEXTANALYTICS_ENDPOINT");
```
Comment (heaths) [inline sdk/textanalytics/Azure.AI.TextAnalytics/perf/Azure.AI.TextAnalytics.Perf/Infrastructure/PerfTestEnvironment.cs:30]: This means having to re-record all sessions - which you did - and git does not store diffs. It stores whole files. For just renaming variables, is growing the repo even further worth it? Maybe a TODO for next time you need to rerecord for a new API version or something?

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsSampleBase.cs [hunk 3]
```diff
@@ -3,17 +3,38 @@
...
             return options;
         }
+
+        internal void IgnoreIfNotPublicCloud()
+        {
+            if (TestEnvironment.GetAudience() != TextAnalyticsAudience.AzurePublicCloud)
+            {
+                Assert.Ignore("Currently, these tests can only be run in the public cloud.");
+            }
+        }
```
Comment (heaths) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsSampleBase.cs:38]: Why not just add this in the `TextAnalyticsBase` from which every test fixture should derive, IIRC?

**Review by heaths — APPROVED** [review]

File: sdk/textanalytics/tests.yml (hunk 1 above)
Comment (joseharriaga) [reply sdk/textanalytics/tests.yml:19]: While it was listed here, there were no tests actually being run against Canary. I've created an issue to follow up on this: https://github.com/Azure/azure-sdk-for-net/issues/35639

File: sdk/textanalytics/Azure.AI.TextAnalytics/perf/Azure.AI.TextAnalytics.Perf/Infrastructure/PerfTestEnvironment.cs (hunk 2 above)
Comment (joseharriaga) [reply sdk/textanalytics/Azure.AI.TextAnalytics/perf/Azure.AI.TextAnalytics.Perf/Infrastructure/PerfTestEnvironment.cs]: Sounds good. I undid the change.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsSampleBase.cs (hunk 3 above)
Comment (joseharriaga) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsSampleBase.cs]: Semantically, it made more sense to me in the test bases, but this does let us avoid the code duplication, so I moved it.

**Review by mssfang — APPROVED** [review]

**Pull request merged**